lib/References/ResolvedReference.php

```diff
@@ -42,4 +52,21 @@ public function getTitles() : array
     {
         return $this->titles;
     }
+
+    /**
+     * @return string[]
+     */
+    public function getAttributes() : array
+    {
+        return $this->attributes;
+    }
+
+    protected function validateAttributes() : void
```

Comment:
`private`

Reply:
Make the attributes a parameter, avoid setting them to property until after calling this method